### ISSUE TITLE
Feature request: API for downloading dataset by id

### ISSUE THREAD

**Issue by amueller** [openml/OpenML#519]
I find the distinction between data API and "normal" API quite confusing, and I'm not sure why a user would ever want to use the data API.

How about we add API calls for both CSV and ARFF to download the data by using the dataset id?
I find it a bit odd that I can get anything about a dataset with a rest call from the API, but I can't get the dataset itself.

**Comment by joaquinvanschoren** [comment]
Indeed, I've been discussing this with @janvanrijn. There are different ways to do this. 

We could integrate the data API in the regular API. The main issue is that then *all* the download urls will change. So you need to flush your cache.

Another solution is to redirect calls from the regular API to the data API, but that seems more work to maintain.

@berndbischl @giuseppec @mfeurer What do you think? Could you live with a one-time change to the download URLs?

@janvanrijn Is it possible to integrate but keep the data API also online? Then we can deprecate it and phase it out?

**Comment by berndbischl** [comment]
we can live with brief, well-communicated changes to the API

**Comment by berndbischl** [comment]
to make sure i am understanding this correctly: 
andi is asking for new API call to get the dataset FILE (by providing the name)?

is that it?

**Comment by joaquinvanschoren** [comment]
Sorry about the confusion. I thought that @amueller was asking two things:
* Integrate the two APIs so that there is 1 endpoint (and eliminate confusion)
* Add a new API call to retrieve a dataset file by data_id.

But maybe adding the new call is sufficient for now? @janvanrijn, what do you think? Can we just do a file_id lookup and then redirect to the download API internally? Note that in that case, training split files etc will still be pulled through the data API, so it only adds a little convenience function. Integrating the APIs would be the more robust solution.

There is also a small caveat: when the dataset is a large dataset that lives elsewhere (e.g. on Zenodo, Figshare,...), and we don't mirror it, we don't really want to stream that through the OpenML server. You should download it directly. I guess that in those cases we could return an error message.

**Comment by amueller** [comment]
Yes, the idea would be to have one consistent API. Right now there are two. The idea was to download a dataset file by dataset ID (not the name, since that is not unique).
Right now downloading any file requires two calls, even if you have the ID: one to match from the (dataset/task/run/...) ID to the file ID, one to download the file. Maybe there are more urgent things right now, though. I guess more documentation would be more important?

**Comment by berndbischl** [comment]
> I guess more documentation would be more important?

upvoting this
